hive/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -163,6 +171,105 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
     }
   }
 
+  @Override
+  public void createNamespace(Namespace namespace, Map<String, String> meta) {
+    Preconditions.checkArgument(!namespace.isEmpty(),
+        "Cannot create namespace with invalid name: %s", namespace.toString());
+    Preconditions.checkArgument(namespace.levels().length == 1,
+        "Cannot support multi part namespace in Hive MetaStore: %s", namespace.toString());
+    try {
+      clients.run(client -> {
+        client.createDatabase(nameSpaceToHiveDb(namespace, meta));
+        return null;
+      });
+
+    } catch (AlreadyExistsException e) {
+      throw new org.apache.iceberg.exceptions.AlreadyExistsException("Namespace %s already exists:",
+            namespace.toString(), e.getMessage());
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to create namespace " + namespace.toString() + " in Hive MataStore", e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to createDatabase(name)" + namespace.toString(), e);
+    }
+  }
+
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) {
+    if (!namespace.isEmpty() && namespace.levels().length != 1) {
```

Comment:
It would be easier to read this by simplifying the check here. It can be `if (namespace.levels().length > 1) { throw ... }`